### ISSUE TITLE
unassigned FC network 

### ISSUE THREAD

**Issue by davidwallis** [HewlettPackard/POSH-HPOneView#606]
I'm trying to do this in a server profile

```
$conn3 = New-OVServerProfileConnection -ConnectionID 3 -ConnectionType FibreChannel -Network Unassigned -PortId "Mezz 3:1-b" -verbose
```

but I get:

```
VERBOSE: [NEW-OVSERVERPROFILECONNECTION] Bound PS Parameters:
Key                   Value
---                   -----
ConnectionID              3
ConnectionType FibreChannel
Network          Unassigned
PortId           Mezz 3:1-b
Verbose                True


VERBOSE: [NEW-OVSERVERPROFILECONNECTION] ParameterSetName: Common
VERBOSE: [NEW-OVSERVERPROFILECONNECTION] Called from: New-VMServerProfileTemplate
VERBOSE: [NEW-OVSERVERPROFILECONNECTION] Verify auth
VERBOSE: [TEST-OVAUTH] Caller: New-OVServerProfileConnection
VERBOSE: [TEST-OVAUTH] Verify Auth for redacted.hostname
VERBOSE: [TEST-OVAUTH] $Appliance is [HPEOneView.Appliance.Connection]
VERBOSE: [TEST-OVAUTH] Received HPEOneview.Appliance.Connection Object:
ConnectionID Name                          UserName      AuthLoginDomain Default
------------ ----                          --------      --------------- -------
1            redacted.hostname Administrator LOCAL           True


The property 'id' cannot be found on this object. Verify that the property exists and can be set.
At \\redacted\path\Documents\WindowsPowerShell\Modules\HPEOneView.630\6.30.2928.3678\HPEOneView.630.psm1:112484 char:9
+         $_conn.id = [Int]$connectionId
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : PropertyNotFound

The property 'functionType' cannot be found on this object. Verify that the property exists and can be set.
At \\redacted\path\Documents\WindowsPowerShell\Modules\HPEOneView.630\6.30.2928.3678\HPEOneView.630.psm1:112496 char:13
+             $_conn.functionType = [String]$ServerProfileConnectionTyp ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : PropertyNotFound

The property 'name' cannot be found on this object. Verify that the property exists and can be set.
At \\redacted\path\Documents\WindowsPowerShell\Modules\HPEOneView.630\6.30.2928.3678\HPEOneView.630.psm1:112500 char:9
+         $_conn.name                = [String]$name
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : PropertyNotFound

The property 'portId' cannot be found on this object. Verify that the property exists and can be set.
At \\redacted\path\Documents\WindowsPowerShell\Modules\HPEOneView.630\6.30.2928.3678\HPEOneView.630.psm1:112501 char:9
+         $_conn.portId              = [String]$portId
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : PropertyNotFound

The property 'ApplianceConnection' cannot be found on this object. Verify that the property exists and can be set.
At \\redacted\path\Documents\WindowsPowerShell\Modules\HPEOneView.630\6.30.2928.3678\HPEOneView.630.psm1:112502 char:9
+         $_conn.ApplianceConnection = $Network.ApplianceConnection
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : PropertyNotFound

The property 'networkUri' cannot be found on this object. Verify that the property exists and can be set.
At \\redacted\path\Documents\WindowsPowerShell\Modules\HPEOneView.630\6.30.2928.3678\HPEOneView.630.psm1:112503 char:9
+         $_conn.networkUri          = $Network.uri
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : PropertyNotFound
    
    
```

any workarounds or is it user error?

**Comment by jullienl** [comment]
Instead of `ConnectionType FibreChannel`, use `ConnectionType FC `and it will work like: 
`$conn3 = New-OVServerProfileConnection -ConnectionID 3 -ConnectionType FC -Network Unassigned -PortId "Mezz 3:1-b"`

**Comment by davidwallis** [comment]
thanks, I tried that:

```You cannot call a method on a null-valued expression.
At Snipped\Documents\WindowsPowerShell\Modules\HPEOneView.630\6.30.2928.3678\HPEOneView.630.psm1:109758 char:26
+                         {$_.Contains($EthernetNetworksUri)}
+                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull

You cannot call a method on a null-valued expression.
At Snipped\Documents\WindowsPowerShell\Modules\HPEOneView.630\6.30.2928.3678\HPEOneView.630.psm1:109758 char:26
+                         {$_.Contains($EthernetNetworksUri)}
+                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull

New-OVServerProfileTemplate : Bandwidth request (2.5 Gb/s) is not valid for connection 3.
At Snipped.ps1:77 char:5
+     New-OVServerProfileTemplate @serverProfileTemplateParams | Wait-O ...
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (AsyncronousTask:String) [New-OVServerProfileTemplate], ServerProfileTemplateResourceException
    + FullyQualifiedErrorId : InvalidOperation,New-OVServerProfileTemplate
```

params for the splat on above are:

```    
$serverProfileTemplateParams = @{
            Name = $name;
            Description = "Server profile template for $($Name) xyz";
            ServerHardwareType = $sht;
            EnclosureGroup = $eg;
            Affinity = "Bay";
            ManageConnections = $True;
            ConnectionsConsistencyChecking = "Minimum";
            Connections = $Conn1, $Conn2, $Conn3, $Conn4;
            ManageBoot = $true;
            BootMode = "UEFIOptimized";
            SecureBoot = "Enabled";
            BootOrder = "SD";
            BootModeConsistencyChecking = "Exact";
            Bios = $true;
            BiosSettings= $biosSettings;
            BiosConsistencyChecking = "Exact";
            ManageIloSettings = $true;
            IloSettingsConsistencyChecking = "Exact";
            IloSettings = $iloSettings;
            HideUnusedFlexNics = $true
    }
```

**Comment by jullienl** [comment]
When a port is unassigned, the cmdlet requires the `RequestedBW` to be set to `0`. 
So try to use:
`$conn3= New-OVServerProfileConnection -Network unassigned -ConnectionID 5 -ConnectionType FC -PortId "Mezz 3:1-b" -RequestedBW 0`

**Comment by davidwallis** [comment]
thanks that gets it in, however hitting what I think is a bug in the module:

```powershell

You cannot call a method on a null-valued expression.
At <snip>\WindowsPowerShell\Modules\HPEOneView.630\6.30.2928.3678\HPEOneView.630.psm1:109758
char:26
+                         {$_.Contains($EthernetNetworksUri)}
+                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull
```

**Comment by ChrisLynchHPE** [comment]
> thanks that gets it in, however hitting what I think is a bug in the module:
> 
> ```powershell
> You cannot call a method on a null-valued expression.
> At <snip>\WindowsPowerShell\Modules\HPEOneView.630\6.30.2928.3678\HPEOneView.630.psm1:109758
> char:26
> +                         {$_.Contains($EthernetNetworksUri)}
> +                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
>     + FullyQualifiedErrorId : InvokeMethodOnNull
> ```

Yes, that is a defect.  Both `New-OVServerProfile` and `New-OVServerProfileTemplate` do not handle unassigned connections propertly, and will error on that improperly evaluated expression.

**Comment by ChrisLynchHPE** [comment]
Also, the validation logic when building the connection helper object omitted `FibreChannel` that Lionel pointed out.

**Comment by ChrisLynchHPE** [comment]
This is fixed in [Release 6.30.3188.1900](https://github.com/HewlettPackard/POSH-HPEOneView/releases/tag/v6.30.3188.1900) and in [PowerShell Gallery](https://www.powershellgallery.com/packages/HPEOneView.630/6.30.3188.1900).

Please verify before this issue is closed.

**Comment by ChrisLynchHPE** [comment]
Closing due to no further activity.  If you wish, feel free to re-open to discuss further.
